subsys/bluetooth/controller/multithreading_lock.h

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+/**
+ * @multithreading_lock.h
+ *
+ * @brief This file defines APIs needed to lock the BLE controller APIs for threadsafe operation.
+ */
+
+#ifndef MULTITHREADING_LOCK_H__
+#define MULTITHREADING_LOCK_H__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <zephyr.h>
+
+/** Macro for acquiring a lock */
+#if IS_ENABLED(CONFIG_BLECTLR_THREADSAFETY)
+#define MULTITHREADING_LOCK_ACQUIRE() \
+	multithreading_lock_acquire(CONFIG_BLECTLR_THREADSAFETY_LOCK_TIMEOUT)
+#else
+#define MULTITHREADING_LOCK_ACQUIRE() (0)
+#endif
+
+/** Macro for acquiring a lock without waiting */
+#if IS_ENABLED(CONFIG_BLECTLR_THREADSAFETY)
+#define MULTITHREADING_LOCK_ACQUIRE_NO_WAIT() \
+	multithreading_lock_acquire(K_NO_WAIT)
+#else
+#define MULTITHREADING_LOCK_ACQUIRE_NO_WAIT() (0)
+#endif
+
+/** Macro for releasing a lock */
+#if IS_ENABLED(CONFIG_BLECTLR_THREADSAFETY)
+#define MULTITHREADING_LOCK_RELEASE() multithreading_lock_release()
+#else
+#define MULTITHREADING_LOCK_RELEASE()
+#endif
+
+/** @brief Try to take the lock while maintaining the specified blocking behavior.
+ *
+ * This API call will be blocked for the time specified by @ref
+ * CONFIG_BLECTLR_THREADSAFETY_LOCK_TIMEOUT and then return error code.
+ *
+ * @param  timeout		Timeout value for locking API.
```

Comment:
Should probably provide a unit to this. Rename the parameter to `timeout_msec` and put the unit in the description?

Reply:
I had done that earlier but changed my mind. I have followed the convention used everywhere in the zephyr repo as well as in nrfxlib. In this case, I will prefer to keep the same convention. I will change the description though.

Reply:
Ok, as long as the units are there in the description.